eZ/Publish/Core/settings/repository.yml

```diff
@@ -149,4 +150,11 @@ services:
 
     ezpublish.signalslot.signal_dispatcher:
         class: %ezpublish.signalslot.signal_dispatcher.class%
+        factory: ["@ezpublish.signalslot.signal_dispatcher.factory", buildSignalDispatcher]
 
+    ezpublish.signalslot.signal_dispatcher.factory:
+        class: %ezpublish.signalslot.signal_dispatcher.factory.class%
+        arguments:
+            - "%ezpublish.signalslot.signal_dispatcher.class%"
+            - "$repository$"
```

Comment:
@alongosz There are examples of different factories for integration tests vs app, for instnace reposiotry. Grab @pspanja on skype if you need some pointers ;)

Side: For later we should try to get rid of this separation to some extent as it causes issues whenever they are out of sync, and is causing us to have duplicated code for several things _(factories and setup of compiler passes)_.